src/binder/inc/bindertracing.h

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// bindertracing.h
+//
+
+#ifndef __BINDER_TRACING_H__
+#define __BINDER_TRACING_H__
+
+class AssemblySpec;
+class PEAssembly;
+
+namespace BinderTracing
+{
+    bool IsEnabled();
+
+    // If tracing is enabled, this class fires an assembly bind start event on construction
+    // and the corresponding stop event on destruction
+    class AssemblyBindEvent
```

Comment:
Nit: I would call this "operation" or "request" - "event" is overloaded - like this "we use multiple events to describe a bind event" - feels wrong.
Also on the first read I was expecting this to represent a single ETW/EventPipe event - which it does not.